### DISCUSSION TITLE
order by clause must not be constant

### DISCUSSION THREAD

**Discussion by xavierzhao** [apache/doris#21017] [A - General / Q&A] [answered]
Recently I upgraded version to Doris from version 1.2.1  to version 1.2.4-1.

The problem is when I execute unit test it dosen't working, it's look like there is a change in the version that cause this problem.

Here is exception message:

![image](https://github.com/apache/doris/assets/1573662/f956a439-ad20-4582-bf38-a336acc3af7a)

and SQL like this:
`
SELECT
	totalNum - lag ( totalNum,
	1,
	0 ) over (
ORDER BY
	sortNum ) AS totalNum
FROM
	(
	SELECT
		sortNum,
		BITMAP_UNION_COUNT (pv) over (
	ORDER BY
		sortNum ) totalNum
	FROM
		(
		SELECT
			ifnull( a.sortNum, b.sortNum ) sortNum,
			BITMAP_UNION ( ifnull( a.pv, b.pv ) ) pv
		FROM
			(
			SELECT
				1 sortNum,
				BITMAP_UNION (DEVICE_ID) AS pv
			FROM
				APP_LOG
			WHERE
				LOG_DATE >= '2023-06-01'
				AND LOG_DATE <= '2023-06-30'
			ORDER BY
				sortNum ) a
		FULL JOIN (
			SELECT
				'2' sortNum,
				BITMAP_UNION (DEVICE_ID) AS pv
			FROM
				APP_LOG
			WHERE
				LOG_DATE < '2023-06-01'
			ORDER BY
				sortNum ) b ON
			a.sortNum = b.sortNum
		GROUP BY
			sortNum
		ORDER BY
			sortNum 
		) c 
	) final
LIMIT 1
`

Any idea.?

**Comment by xavierzhao** [comment]
@yagagagaga Could you help me, please?

**Reply by yagagagaga** [reply]
This is a bug and 2.0 has fix it. Here is the solution that you maybe avoid this issue:
```sql
CREATE TABLE `tmp` (
  `sortNum` int(11) NULL COMMENT "",
  `pv` bitmap BITMAP_UNION NULL COMMENT ""
) ENGINE=OLAP
AGGREGATE KEY(`sortNum`)
COMMENT "OLAP"
DISTRIBUTED BY HASH(`sortNum`) BUCKETS 1
PROPERTIES (
  "replication_num" = "1"
);

insert into tmp
SELECT
                    ifnull(a.sortNum, b.sortNum) sortNum,
                    BITMAP_UNION (ifnull(a.pv, b.pv)) pv
                FROM
                    (
                        SELECT
                            1 sortNum,
                            BITMAP_UNION (DEVICE_ID) AS pv
                        FROM
                            APP_LOG
                        WHERE
                            LOG_DATE >= '2023-06-01' AND LOG_DATE <= '2023-06-30'
                        ORDER BY
                            sortNum
                    ) a FULL JOIN (
                        SELECT
                            '2' sortNum,
                            BITMAP_UNION (DEVICE_ID) AS pv
                        FROM
                            APP_LOG
                        WHERE
                            LOG_DATE < '2023-06-01'
                        ORDER BY
                            sortNum
                    ) b ON a.sortNum = b.sortNum
                GROUP BY
                    sortNum
                ORDER BY
                    sortNum;

SELECT
    totalNum - lag (totalNum, 1, 0) over (ORDER BY sortNum) AS totalNum
FROM
    (
        SELECT
            sortNum,
            BITMAP_UNION_COUNT (pv) over (ORDER BY sortNum ) totalNum
        FROM
            tmp
    ) final
LIMIT
    1;
```

**Reply by xavierzhao** [reply]
I have many query criteria and temporary tables are not suitable for handling this issue. Do you have any plans to release a new version of 1.0 or a stable version of 2.0 in the future? @yagagagaga

**Reply by yagagagaga** [reply]
We have added this case to the Regression testing, and I guarantee that this problem will not occur again in the future. Doris 2.0 will be released around mid July. You can wait just a few days.

**Reply by xavierzhao** [reply]
Is 1.0 no longer maintained?  @yagagagaga

**Reply by yagagagaga** [reply]
Maybe it will fix at 1.2.6. But I'm not sure when this version will be released.

**Reply by xavierzhao** [reply]
@yagagagaga Does the recently released 1.2.6 version fix this issue?

**Reply by yagagagaga** [reply, marked as answer]
This PR [#21820](https://github.com/apache/doris/pull/21820) hasn't merge yet.
